cli/src/action/registry/api.rs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        request
+            .send()
+            .map_err(|err| CliError::ActionError(format!("Failed to add node to registry: {}", err)))
+            .and_then(|res| {
+                let status = res.status();
+                if status.is_success() {
+                    Ok(())
+                } else {
+                    let message = res
+                        .json::<ServerError>()
+                        .map_err(|_| {
+                            CliError::ActionError(format!(
+                                "Registry add node request failed with status code '{}', but error response was not valid",
+                                status
+                            ))
+                        })?
+                        .message;
+
+                    Err(CliError::ActionError(format!(
+                        "Failed to add node to registry: {}",
+                        message
+                    )))
+                }
+            })
+    }
+
+    /// Retrieves the node with the given identity from the registry.
+    pub fn get_node(&self, identity: &str) -> Result<Option<NodeResponse>, CliError> {
+        // Allowing unused_mut because request must be mutable if experimental feature
+        // splinter-cli-jwt is enabled, if feature is removed unused_mut notation can be removed
+        #[allow(unused_mut)]
+        let mut request = Client::new().get(&format!("{}/registry/nodes/{}", self.url, &identity));
+
+        #[cfg(feature = "splinter-cli-jwt")]
+        {
+            request = request.header("Authorization", &self.auth);
+        }
+
+        request.send()
+            .map_err(|err| CliError::ActionError(format!("Failed to fetch node: {}", err)))
+            .and_then(|res| {
+                let status = res.status();
+                if status.is_success() {
+                    res.json::<NodeResponse>().map(Some).map_err(|_| {
+                        CliError::ActionError(
+                            "Request was successful, but received an invalid response".into(),
+                        )
+                    })
+                } else {
+                    let message = res
+                        .json::<ServerError>()
+                        .map_err(|_| {
+                            CliError::ActionError(format!(
+                                "Registry get node request failed with status code '{}', but error response was not valid",
+                                status
+                            ))
+                        })?
+                        .message;
+
+                    Err(CliError::ActionError(format!(
+                        "Failed to fetch node: {}",
+                        message
+                    )))
+                }
+            })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NodeResponse {
+    pub identity: String,
+    pub endpoints: Vec<String>,
+    pub display_name: Option<String>,
```

Comment:
I think the display name is always set in the response.

Reply:
fixed